db/python/tables/cohort.py

```diff
@@ -0,0 +1,198 @@
+# pylint: disable=too-many-instance-attributes
+import dataclasses
+import datetime
+
+from db.python.tables.base import DbBase
+from db.python.tables.project import ProjectId
+from db.python.utils import GenericFilter, GenericFilterModel, to_db_json
+from models.models.cohort import Cohort, CohortTemplateModel
+
+
+@dataclasses.dataclass(kw_only=True)
+class CohortFilter(GenericFilterModel):
+    """
+    Filters for Cohort
+    """
+
+    id: GenericFilter[int] | None = None
+    name: GenericFilter[str] | None = None
+    author: GenericFilter[str] | None = None
+    template_id: GenericFilter[int] | None = None
+    timestamp: GenericFilter[datetime.datetime] | None = None
+    project: GenericFilter[ProjectId] | None = None
+
+
+@dataclasses.dataclass(kw_only=True)
+class CohortTemplateFilter(GenericFilterModel):
+    """
+    Filters for CohortTemplate
+    """
+
+    id: GenericFilter[int] | None = None
+    name: GenericFilter[str] | None = None
+    description: GenericFilter[str] | None = None
+    criteria: GenericFilter[dict] | None = None
+    project: GenericFilter[ProjectId] | None = None
+
+
+class CohortTable(DbBase):
+    """
+    Capture Cohort table operations and queries
+    """
+
+    table_name = 'cohort'
+    common_get_keys = [
+        'id',
+        'name',
+        'template_id',
+        'description',
+        'author',
+        'project',
+    ]
+
+    template_keys = [
+        'id',
+        'name',
+        'description',
+        'criteria',
+        'project'
+    ]
+
+    async def query(self, filter_: CohortFilter):
+        """Query Cohorts"""
+        wheres, values = filter_.to_sql(field_overrides={})
+        if not wheres:
+            raise ValueError(f'Invalid filter: {filter_}')
+
+        common_get_keys_str = ','.join(self.common_get_keys)
+        _query = f"""
+        SELECT {common_get_keys_str}
+        FROM cohort
+        WHERE {wheres}
```

Comment:
Indeed this needs considering: I was just writing test code using this `query()` and realised that the Cohorts returned have a `sequencing_groups` field, but it will always be empty…

I guess the idea is to use it in conjunction with `get_cohort_sequencing_group_ids()`, but the returned empty field is still a bit of a surprise.

Reply:
Ok I've updated the typehints here to help with clarity. This returns a `CohortInternal` object, which does not expect sequencing_groups in the `from_db` function. 

sequencing_groups are a field of a `GraphQLCohort` in the schema.py file, (as John said, through the get_cohort_sequencing_group_ids() function) and are returned that way. 

What's the context where you're getting an empty field @jmarshall? I might have misinterpreted something here.